### DISCUSSION TITLE
escapeHtml does't work properly in  react-markdown

### DISCUSSION THREAD

**Discussion by meizibupt** [remarkjs/remark#576] [Q&A] [answered]
looks like escapeHtml is enabled by default. If my input has special character in the input string, it renders the special character instead of rending the original string. 

For example, if the string is "This is euro & #8364;", when it renders, it renders "This is euro  €"

**Comment by meizibupt** [comment]
Is there a way to keep showing the special character encoding in the output rather than converting it into the real character?

**Comment by ChristianMurphy** [comment, marked as answer]
> looks like escapeHtml is enabled by default. If my input has special character in the input string, it renders the special character instead of rending the original string.
> "This is euro & #8364;", when it renders, it renders "This is euro €"

That's because it isn't HTML specific, that is how markdown works
https://spec.commonmark.org/dingus/?text=%26%238364%3B%0A
and also why you had to add a space in GitHub, because GitHub flavored markdown builds on CommonMark.

It sounds like what you are really after is an inline code block
```markdown
This is euro `&#8364;`
```
or an escape character

```markdown
This is euro \&#8364;
```

**Reply by meizibupt** [reply]
Thanks. I guess my question can extend to if I have a user input string and want to put it inside a markdown string,  is there a way I can ignore any markdown format for the user input string?

**Reply by ChristianMurphy** [reply]
>  if I have a user input string and want to put it inside a markdown string

What do you mean by this?
Typically the user/author enters all the text, so the user input string is the markdown string.
Are you doing something different? Is there a distinction in your use case?

**Reply by meizibupt** [reply]
The use case is we have some user input field that user can enter free texts (comments for example), and we want to put the user input within a markdown context, something like this  `[markdown string]+userInput+[markdown string]`, whatever the user input is we want to keep it as is and don't treat as markdown string.

**Reply by ChristianMurphy** [reply]
That isn't something which can be easily solved with just text.
You could use a codeblock as suggested above or a [directive](https://github.com/remarkjs/remark-directive), the problem with both (and any text based approach) is that users could accidentally (or intentionally) escape out of that block and write markdown again.
If you want to ensure that the user can write text and only text, would be to work with remark directly, and add the user content as a text node in the AST. https://unifiedjs.com/learn/ offers more insights into working with ASTs.

**Reply by wooorm** [reply]
Or: why not use three things? For the initial and final markdown, use react-markdown. For the user content in between, use a different element?
